jax/lax/lax_control_flow.py

```diff
@@ -536,18 +537,20 @@ def _scan_jvp(primals, tangents, forward, length, jaxpr, num_consts, num_carry,
   nonzeros = [t is not ad_util.zero for t in tangents]
   const_nz, init_nz, xs_nz = split_list(nonzeros, [num_consts, num_carry])
 
+  # Fixpoint computation of which carry are not ad.zero: either
+  # non-zero from init, or the carry out is non-zero. Each iteration promotes
+  # at least one carry to non-zero.
   carry_nz = init_nz
-  for _ in range(1000):
+  while True:
     nonzeros = const_nz + carry_nz + xs_nz
     jaxpr_jvp, nonzeros_out = ad.jvp_jaxpr(
         jaxpr, nonzeros, instantiate=carry_nz + [False] * num_ys)
     carry_nz_out, ys_nz = nonzeros_out[:num_carry], nonzeros_out[num_carry:]
     if carry_nz_out == carry_nz:
       break
     else:
-      carry_nz = carry_nz_out
-  else:
-    raise FixedPointError
+      carry_nz = _map(operator.or_, carry_nz, carry_nz_out)
```

Comment:
This explicit lattice join is unnecessary when we pass in `instantiate=carry_nz + ...` (which I believe is required, as per my other comment). Moreover, we need the `jaxpr_jvp` produced to match the signature of `carry_nz`, and doing a lattice join here wouldn't update the jaxpr to match.

Reply:
I added the lattice join because it makes it locally-obvious that this loop will terminate. (The same goal would be achieved with an assert.) After the join we do at least one more iteration, to get the jaxpr right.

Reply:
You're right: it's not locally obvious that it will terminate as written on master, since one needs to understand that `instantiate` is effectively doing the lattice join.

Reply:
I don't think this has an extra iteration because the loop termination condition is exactly as before. Only if the carry_out has changed, we do the OR and then iterate again.